libgadget/blackhole.c

```diff
@@ -540,6 +541,21 @@ blackhole(const ActiveParticles * act, double atime, Cosmology * CP, ForceTree *
     tw_accretion->haswork = NULL;
     tw_dynfric->haswork = NULL;
 
+    /* Types used in treewalks:
+     * dynamical friction uses: stars, DM if BH_DynFrictionMethod > 1 gas if BH_DynFrictionMethod  == 3.
+     * accretion uses: all types but ONLY for repositioning potential minimum. Otherwise gas + black holes. gas + stars + BH is probably fine.
+     * gas + BH probably not enough if gas is sparse in halo.
+     * feedback uses: gas + black holes.
+     * The DM in dynamic friction and accretion doesn't really do anything, so could perhaps be removed from the treebuild later.
+     * However, we would still need a tree with gas + DM in the wind code.
+     */
+    if(!tree->tree_allocated_flag)
+    {
+        message(0, "Building tree in blackhole\n");
+        force_tree_rebuild_mask(tree, ddecomp, GASMASK + DMMASK + STARMASK + BHMASK, 0, NULL);
```

Comment:
"|"?

Reply:
Sure, I suppose. I was following the style we use for the .mask in ngbiter.